tools/querytee/proxy_endpoint.go

```diff
@@ -386,3 +408,100 @@ func (r *backendResponse) statusCode() int {
 
 	return r.status
 }
+
+// extractMinTimeFromRequest extracts the minimum time from a Prometheus API request
+// Returns zero time if time parameters cannot be parsed, which causes all backends to handle the query.
+func extractMinTimeFromRequest(r *http.Request) time.Time {
+	if err := r.ParseForm(); err != nil {
```

Comment:
Trying out this new feature, I just realized this is problematic as it may read the body. And when forwarding the request we'd want to read it again. This produces the following error:

```
err="executing backend request: Post \"http://***/prometheus/api/v1/query_range\": http: Request.ContentLength=169 with nil Body"
```


I was able to reproduce it with a test that forces the `ParseForm` to read the body. Here's the diff, with an example fix: https://github.com/grafana/mimir/compare/jvp/query-tee-extend-backend-config-json...njpm/double-read-bug . The fix is just a draft

Reply:
I've cherry picked your commits onto this branch. Thanks Nico